### PR TITLE
Issue#363

### PR DESCRIPTION
**Pull request by bmpj13**
Tackling [issue#363](https://github.com/shouldly/shouldly/issues/363) . The idea is to create Shouldly's own TimeoutException, so it doesn't get mixed with the TimeoutException provided by .NET Framework.

### CODE REVIEW

**Comment by bmpj13** [comment]
Sorry guys, I closed the [other PR](https://github.com/shouldly/shouldly/pull/413) because I accidentally commited a file that wasn't supposed to be there :smile:

File: src/Shouldly/ShouldlyTimeoutException.cs [hunk 1]
```diff
@@ -0,0 +1,15 @@
+﻿using System;
+
+namespace Shouldly
+{
+    public class ShouldlyTimeoutException : Exception
```
Comment (JosephWoodward) [inline src/Shouldly/ShouldlyTimeoutException.cs]: Would it be better to extend `TimeoutException` in this instance?

Comment (JosephWoodward) [inline src/Shouldly/ShouldlyTimeoutException.cs]: What effect (if any) would this have on backwards compatibility? (@JakeGinnivan may be able to answer this as he knows everything)

Comment (bmpj13) [inline src/Shouldly/ShouldlyTimeoutException.cs]: After some testing, I think it, in fact, would be better to extend TimeoutException. Not only for backwards compatibility, but also to enhance Shouldly's messages.


The following code
```
[Test]
public void ShouldHandleTimeoutException()
{
    var task = new Func<Task>(() => {System.Threading.Thread.Sleep(20000); 
                                     return new Task(null); });
    task.ShouldThrow(typeof(Exception));
}
```

With my code claims that "Shouldly.ShouldlyTimeoutException was thrown". 
When extending TimeoutException it says that the operation exceeded the time limit, which is clearer.

Comment (JosephWoodward) [inline src/Shouldly/ShouldlyTimeoutException.cs]: Sounds better 👍

**Comment by bmpj13** [comment]
@JosephWoodward I don't know why it is failing. The test *FuncOfTaskOfTWithTimeoutScenario* passes in my machine... Can you take a look at it?

**Comment by bmpj13** [comment]
Ok so apparently it had to do with the .NET version... all good now :+1:

**Comment by JakeGinnivan** [comment]
LGTM, :shipit:

**Comment by JosephWoodward** [comment]
Thanks for this @bmpj13! Merged 👍

**Pull request merged**